bvm/ballerina-runtime/src/test/java/io/ballerina/runtime/test/config/negative/TOMLProviderNegativeTest.java

```diff
@@ -0,0 +1,244 @@
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.runtime.test.config.negative;
+
+import io.ballerina.runtime.api.Module;
+import io.ballerina.runtime.api.PredefinedTypes;
+import io.ballerina.runtime.api.creators.TypeCreator;
+import io.ballerina.runtime.api.flags.SymbolFlags;
+import io.ballerina.runtime.api.types.ArrayType;
+import io.ballerina.runtime.api.types.Field;
+import io.ballerina.runtime.api.types.IntersectionType;
+import io.ballerina.runtime.api.types.MapType;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.TableType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.internal.configurable.ConfigResolver;
+import io.ballerina.runtime.internal.configurable.VariableKey;
+import io.ballerina.runtime.internal.configurable.providers.toml.TomlFileProvider;
+import io.ballerina.runtime.internal.diagnostics.DiagnosticLog;
+import io.ballerina.runtime.internal.types.BIntersectionType;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.ballerina.runtime.test.TestUtils.getConfigPathForNegativeCases;
+
+/**
+ * Test cases specific for configuration provided via TOML files/content.
+ */
+public class TOMLProviderNegativeTest {
+
+    private final Module module = new Module("myOrg", "test_module", "1.0.0");
+    private final Module subModule = new Module("myOrg", "test_module.util.foo", "1.0.0");
+
+    @Test(dataProvider = "path-test-provider")
+    public void testPathErrors(String tomlFileName, String errorMsg, int warningCount) {
+        DiagnosticLog diagnosticLog = new DiagnosticLog();
+        VariableKey intVar = new VariableKey(module, "byteVar", PredefinedTypes.TYPE_BYTE, true);
+        ConfigResolver configResolver =
+                new ConfigResolver(module, Map.ofEntries(Map.entry(module, new VariableKey[]{intVar})), diagnosticLog,
+                        List.of(new TomlFileProvider(getConfigPathForNegativeCases(tomlFileName))));
+        configResolver.resolveConfigs();
+        Assert.assertEquals(diagnosticLog.getErrorCount(), 1);
+        Assert.assertEquals(diagnosticLog.getWarningCount(), warningCount);
+        Assert.assertTrue(diagnosticLog.getDiagnosticList().get(0).toString().matches(errorMsg));
```

Comment:
when using `assertTrue` better to provide a message, otherwise fails are harder to narrow down.